azure-kusto-data/azure/kusto/data/security.py

```diff
@@ -0,0 +1,67 @@
+"""A module contains all kusto security related classes."""
+
+from datetime import timedelta, datetime
+from enum import Enum
+import webbrowser
+import dateutil.parser
+from adal import AuthenticationContext
+from azure.kusto.data import KustoConnectionStringBuilder
+from azure.kusto.data.exceptions import KustoClientError
+
+
+class AuthenticationMethod(Enum):
+    """Enum represnting all authentication methods available in Kusto."""
+    aad_username_password = "aad_username_password"
+    aad_application_key = "aad_application_key"
+    aad_device_login = "aad_device_login"
+
+class _AadHelper(object):
+    def __init__(self, kcsb):
+        try:
+            authority = kcsb.authority_id or 'microsoft.com'
+        except KeyError:
```

Comment:
Why would `kcsb` raise a `KeyError` on an attribute access? Tests should have caught this.